### ISSUE TITLE
[A11Y][PIN code] Focus should not move automatically to the next field

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-flutter#649]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

Samsung and iOS

### OS Version

.

### App version

1.2.0

### Describe the issue

Once a digit is entered, the focus should not move automatically to the next field

### Expected Behavior

Keep the focus on the current field and let the user move it by himself/herself

**Comment by AhmedAmineZr** [comment]
@pya35 this functionality is recommended from @B3nz01d and we have this ticket #555  to be corrected but describe that this is the behavior that we needed :) 
we need to made it in delete case when we delete a last field we need to switch from field to another

**Comment by pya35** [comment]
@AhmedAmineZr @B3nz01d see my comment https://github.com/Orange-OpenSource/ouds-flutter/issues/555#issuecomment-4223576692

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@AhmedAmineZr ) to get access.

Display name: DesignToolbox (ALPHA) (696)
Version: 2.0.0
Build type: ALPHA
Build details (GitHub): 691
Related issues: 649, 628
cc @AhmedAmineZr @nouha06 (dev team)
cc @pya35 (accessibility team)

**Comment by nouha06** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@AhmedAmineZr) to get access.

Display name: Design Toolbox Flutter (696)
Version: 2.0.0
Build type: Alpha
Build details (GitHub): 691
Related issues: #628, #649
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)

**Comment by AhmedAmineZr** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@AhmedAmineZr ) to get access.

Display name: DesignToolbox (ALPHA) (703)
Version: 2.0.0
Build type: Alpha
Build details (GitHub): 691
Related Issues: #791 #776 #649 #628
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @pya35 (Accessibility)

**Comment by AhmedAmineZr** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@AhmedAmineZr) to get access.

Display name: Design Toolbox Flutter (703)
Version: 2.0.0
Build type: Alpha
Build details (GitHub): 691
Related Issues: #791 #776 #649 #628
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @pya35 (Accessibility)

**Comment by B3nz01d** [comment]
@pya35 are we sure this is the right implementation we want to provide to the user?
When I test this updated implementation the user needs to:
* select a field
* in the keyboard select the right value
* exit the keyboard
* select the next field
* in the keyboard select the right value
* ...
There seems to be a lot of extra steps that will make this field very difficult to use. Is there any other option possible?

**Comment by pya35** [comment]
You're right @B3nz01d
@AhmedAmineZr @nouha06 After doing some tests with both ways of navigation, I'm agree that it's quite difficult to fill in the PIN code when focus is not moving automatically. The best solution is to move the focus on next input automatically as done previously.

But there's several issues for this case (tested on build 704) 

- [x] when focus moves to the next field, the new focused input is not vocalized: this field shall be vocalized to inform the user
- [x]  the current input name is "code à chiffres 1, code à chiffre 2...", so if we fill in a digit we hear for example "code à chiffres 1, 5", so it can be confusing bewteen the position and the value: we should rename the fields like "1er chiffre, 2eme chiffre..." / "1st digit, 2nd digit...." 
- [x]  inputs are grouped with a global name "entrez votre code à 6 chiffres, double tap pour modifier" but the double-tap doesnt work : allow double-tap to edit the first digit
- [x] when user deletes an input, the focus goes back to the previous field: the focus should stay in the deleted input

**Comment by pya35** [comment]
@nouha06 Tested on build 708 on iOS with VoiceOver:
- when focus moves to the next field, the new focused input is vocalized "chiffre 2 sur 6, chiffre 3 sur 6..." which is different from individual input labels: we should keep the same label used for each digit: "1er chiffre, 2eme chiffre..." / "1st digit, 2nd digit...."
- inputs are grouped with a global name "entrez votre code à 6 chiffres": now we can double-tap to edit, good, but the hint as been removed", we should keep it to inform the user. 
- when user deletes an input, the focus goes back to the previous field: the focus should stay in the deleted input
- the edition of each individual input doesn't work, it's only possible to edit the first empty digit, so we can't change/delete a previous digit without deleting next ones.

**Comment by pya35** [comment]
@AhmedAmineZr @nouha06 FYI, rules above are also applicable to keyboard navigation
